hail/src/main/scala/is/hail/expr/ir/BlockMatrixIR.scala

```diff
@@ -563,7 +663,14 @@ case class BlockMatrixSparsify(
   value: IR,
   sparsifier: BlockMatrixSparsifier
 ) extends BlockMatrixIR {
-  def typ: BlockMatrixType = child.typ
+
+  //FIXME: We should probably stop generating IR for the value; as discussed,
```

Comment:
what does this change involve? Easy to do as a quick follow-up?

Reply:
I'm not sure, and I think the easiest way to change this would be to make `value` a Scala literal and move it onto the sparsifier. The only problem with this is that we currently rely on being able to pass IR in ld_matrix to avoid serializing the (large) intervals to/from py4j, but that's something we could solve separately.

Reply:
yeah, I'm quite convinced we can't do this. It will compile and evaluate every time the node is constructed/copied. Could do it in parse, maybe, to have the same from-python semantics?

Reply:
sounds good! I can do something like that here, so that it doesn't affect existing stuff when it goes it, and then look at moving it someplace more reasonable? (currently should only affect `ld_matrix`, since everywhere else we're compiling and evaluating a `Literal` IR, but that still seems pretty bad)

Reply:
yep, sounds fine to fix python later!

Reply:
ok, done! I tried to keep the mess confined to ir.Parser so it'll be easy to go in and clean up later :)